frontend/src/App/App.tsx

```diff
@@ -7,37 +7,52 @@ import PlusButton from "../plusButton/PlusButton";
 import Search from "../Search/Search";
 import MuralForm from "../muralForm/MuralForm";
 import SigninForm from "../SignInForm/SigninForm";
-import Context from "../context"
-import "firebase/auth"
-import FirebaseAuth from "../firebase"
+import Context from "../context";
+import "firebase/auth";
+import FirebaseAuth from "../firebase";
 
 function App() {
   const [isSignedIn, setIsSignedIn] = useState<boolean>(false);
-  const [signingIn, setSigningIn] = useState<boolean>(false)
-  const [user, setUser] = useState<any>({})
+  const [signingIn, setSigningIn] = useState<boolean>(false);
+  const [user, setUser] = useState<any>({});
   const [sidebarOpen, setSidebarOpen] = useState<boolean>(false);
   const [murals, setMurals] = useState<any>([]);
+  const [signInError, setSignInError] = useState<string>('');
 
   const handleSignin = (creds: any) => {
+    setSignInError("")
     FirebaseAuth.signInWithEmailAndPassword(creds.email, creds.password)
-      .catch((error: any) => console.log(error.message));
-  }
+      .then(() => {
+        handleCancelSignin()
+        setSignInError('')
+      })
+      .catch((error: any) => {
+        console.log(error.message);
+        if (error.code === "auth/network-request-failed") {
+          setSignInError("Network error.")
+        } else if (error.code === 'auth/wrong-password') {
+          setSignInError("Incorrect password.")
+        } else if (error.code === "auth/user-not-found") {
+          setSignInError("No user record corresponding to this email.")
```

Comment:
nice error messages!

these could be combined to a switch statement but it's fairly readable now

Reply:
They both work
![Screen Shot 2021-01-27 at 7 30 04 PM](https://user-images.githubusercontent.com/59661782/105985194-1e25e580-60d6-11eb-8bf9-91da52da45e3.png)
![Screen Shot 2021-01-27 at 7 30 19 PM](https://user-images.githubusercontent.com/59661782/105985218-24b45d00-60d6-11eb-87ff-fd2562dbb28b.png)